### PR TITLE
Add tests, refactor a smidge

### PR DESCRIPTION
**Pull request by maxjacobson**
(some more info in individual commit messages)

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,5 +1,9 @@
 use std::io;
 
+fn count_overlap(secret: &str, guess: &str) -> usize {
+    secret.chars().filter(|letter| guess.contains(*letter)).count()
```
Comment (maxjacobson) [inline src/main.rs:4]: (note: I have no idea why this `*` is necessary but it doesn't work without it 🤷‍♂️ )

Comment (ABaldwinHunter) [reply src/main.rs:4]: I like pulling this functionality out into a separate function. Nice use of rust built-in functions.

Just slightly curious if this new function is more performant though?

Comment (maxjacobson) [reply src/main.rs:4]: @ABaldwinHunter in fact it's less performant, by 6 nanoseconds https://github.com/ABaldwinHunter/gelato/pull/2/commits/42a258d38bd41fe2e2cfa9769c73ccf23b3d819b 🤔

Comment (maxjacobson) [reply src/main.rs:4]: well I just tried again and now the results are kinda flip flopped, so I dunno... I think the difference is small enough that going with the more "nice" version is worth doing. WDYT?

**Comment by ABaldwinHunter** [comment]
LGTM :)

**Pull request merged**